index.js

```diff
@@ -602,6 +635,81 @@ function packAddress(derivationPath, xpub, hdPassphrase, derivationScheme) {
   )
 }
 
+function getAddressHeader(addressType, networkId) {
+  return Buffer.from([(addressType << 4) | networkId])
+}
+
+function getSpendingPart(pubKey) {
+  return blake2b(pubKey, 28)
+}
+
+function deriveBaseAddress(pubKey, stakePubKey, addressType, networkId, isStakeHash = false) {
+  validateBuffer(pubKey, 32)
+  validateAddressType(addressType, 0, 3)
```

Comment:
it's tempting to validate the address type against a range, but this kind of leaks a low-level implementation detail of the protocol and puts a potentially unnecessary burden on whoever tries to understand it. So I think it would be more idiomatic to validate it explicitly against the set of allowed values which is not that big after all